spec/javascripts/components/search-autocomplete-spec.js

```diff
@@ -0,0 +1,175 @@
+/* eslint-env jasmine */
+/* global GOVUK, KeyboardEvent */
+
+describe('Search autocomplete component', function () {
```

Comment:
i am treating fetch request related stuff as a succinct thing. everything else here i think addressed. testing keyup i feel is heavily implicity tested through out the tests so i am not sure it needs testing succinctly?

Reply:
all tests now use XMLHttpRequest instead of fetch. it seems that XMLHttpRequest is used in existing components, probably because Jasmine cant handle fetch without use of polyfills. so now Jasmine mocks http responses.

Reply:
@kevindew a bit going here apart from what i already mentioned but i think ive addressed comments just across a few commits

Reply:
Ah the reason the other components use XMLHttpRequest is because they was designed to work with IE11, we should use the modern fetch API here over that.

We've got precedence for testing fetch with Jasmine: https://github.com/alphagov/govuk-chat/blob/30f5d8bf1e0541e7b65632955a5952971cf3f604/spec/javascripts/modules/chat-conversation-spec.js#L163-L166